docs/rfcs/2024-08-06-json-datatype.md

```diff
@@ -0,0 +1,120 @@
+---
+Feature Name: Json Datatype
+Tracking Issue: https://github.com/GreptimeTeam/greptimedb/issues/4230
+Date: 2024-8-6
+Author: "Yuhan Wang <profsyb@gmail.com>"
+---
+
+# Summary
+This RFC proposes a method for storing and querying JSON data in the database.
+
+# Motivation
+JSON is widely used across various scenarios. Direct support for writing and querying JSON can significantly enhance the database's flexibility.
+
+# Details
+
+## Storage and Query
+
+The type system of GreptimeDB is based on the types of arrow/datafusion, each type has a corresponding physical type from arrow/datafusion. Thus, the json type is built on top of the `Binary` type, utilizing current implementation of both `Value` and `Vector` of it. JSON type performs the same as Binary type inside the storage layer and query engine.
+
+This also brings 2 problems: insertion and query interface.
+
+## Insertion
+
+User commonly write JSON data as strings. Thus we need to make conversion between string and binary data. There are 2 ways to do this:
+
+1. MySQL and PostgreSQL servers provide auto-conversion between string and JSON data. When a string is inserted into a JSON column, the server will try to parse the string as JSON data and convert it to binary data of JSON type. The non-JSON string will be rejected.
```

Comment:
```suggestion
1. MySQL and PostgreSQL servers provide auto-conversion between string and JSON data. When a string is inserted into a JSON column, the server will try to parse the string as JSON data and convert it to binary data of JSON type(JSONB). The non-JSON string will be rejected.
```

Reply:
Can you explain the difference between SQL insertion and inserting through the prepared statement? You can draw a simple graph to explain the internal details.